### DISCUSSION TITLE
Jacobian from the isoparametric formulation

### DISCUSSION THREAD

**Discussion by HoseongJeong** [kinnala/scikit-fem#1163] [Q&A] [answered]
Hello,

I would like to express my sincere gratitude to the scikit-fem team for all your hard work.

I have a question regarding the bilinear form. Should the Jacobian arising from the isoparametric formulation be explicitly included when defining the bilinear form?

Thank you for your kind assistance despite your busy schedule.

**Comment by kinnala** [comment, marked as answer]
Within the bilinear form all quantities are global and mapped through the isoparametric mapping before passed onto the form function provided by the user. In this case, it is the responsibility of `skfem.mapping.MappingIsoparametric` to transform the local gradients using the Jacobian of the isoparametric formulation and the user does not need to interfere.

**Reply by HoseongJeong** [reply]
Thanks for your answer kinnala!!
